crates/task-impls/src/transactions.rs

```diff
@@ -231,25 +262,38 @@ impl<
                 continue;
             };
 
-            let mut available_blocks = match self
-                .builder_client
-                .get_available_blocks(
+            let mut available_blocks = match async_compatibility_layer::art::async_timeout(
+                self.api.propose_max_round_time(),
```

Comment:
Maybe tighten it to `self.api.propose_max_round_time() - task_start_time.elapsed()`?